### PR TITLE
feat: Enable drag and drop of event types

### LINKED ISSUE #10016: Allow drag and drop to rearrange event types
### Is your proposal related to a problem?

It would be nice if we could drag and drop the event types instead of using arrows to move them one step up or down.

https://github.com/calcom/cal.com/assets/47350983/71919323-eeba-4828-8ec5-1e95414f35b8

### Describe the solution you'd like

Implement drag and drop using [react-beautiful-dnd](https://www.npmjs.com/package/react-beautiful-dnd?activeTab=readme).

### Describe alternatives you've considered

There are other libraries like react-dnd but react-beautiful-dnd seems to be the most popular one.

### Additional context

There was an issue that was raised before and was closed as it was low priority and since drag and drop tends to be flaky: https://github.com/calcom/cal.com/issues/7125.
With this in mind this feature can be considered a prototype, which will need extensive testing before it can be submitted.

### PR DESCRIPTION
**Pull request by Praneeth2k**
## What does this PR do?

Fixes #10016
/claim #10016

https://github.com/calcom/cal.com/assets/47350983/6a4ab7da-e4d7-47d8-b8f7-e5f058905f58

https://github.com/calcom/cal.com/assets/47350983/a4881ee8-362d-42bf-b687-fda2bfd855af

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

In the event-types page, move events around by dragging and dropping them. Click on refresh to ensure that the order is preserved. Works with mobile as well.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Praneeth2k** [comment]
@zomars I would like to work on this feature and document the challenges with implementing drag and drop. If I am able to overcome the roadblocks, we could consider integrating this with main. I am still working on this and am yet to test this out extensively, but was able to implement drag and drop (have attached the video in the description) and have started noticing some flaky behaviour when I move the events too fast.

**Comment by PeerRich** [comment]
reminder we only review non-draft PRs

**Comment by PeerRich** [comment]
whats missing to leave draft?

**Comment by PeerRich** [comment]
/bonus 10 can you revisit this PR and get it over the finish line? @Praneeth2k

**Comment by Praneeth2k** [comment]
@PeerRich, Sure, will post an update by 6th this month.

**Comment by Praneeth2k** [comment]
I've opened this PR for review. Had run into some issues while moving around the items too fast but they were fixed automatically once I pulled in the latest changes from main. Drag and drop is working fine for both web and mobile interfaces. Order is being preserved upon refreshing.
cc: @PeerRich

File: package.json [hunk 1]
```diff
@@ -95,6 +96,7 @@
     "city-timezones": "^1.2.1",
     "eslint": "^8.34.0",
     "lucide-react": "^0.171.0",
+    "react-beautiful-dnd": "^13.1.1",
```
Comment (O4epegb) [inline package.json:99]: react-beautiful-dnd is basically on life support, and did't have any new features for around 3 years now, maybe it's worth consider something more modern and maintained

**Comment by PeerRich** [comment]
react-beautiful-dnd not really maintained anymore

**Comment by moinbukhari** [comment]
I would recommend using dnd-kit ( https://dndkit.com/) looks like it is better maintained.

**Comment by PeerRich** [comment]
i thought about this more and i dont think we should be doing drag and drop here. sorry about the bounty. will still tip for the work!

**Comment by Praneeth2k** [comment]
No worries! Thanks for the reward

**Pull request closed without merging**